docs/RFC-zero-waste.md

```
@@ -0,0 +1,33 @@
+# RFC: Efficient Scoring Engine & Early Fit Gate
+
+## 1. Goal Description
```

markdownlint-cli2 (0.22.1)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)